transports/httpclient/src/main/java/org/commonjava/maven/galley/transport/htcli/internal/HttpDownload.java

```diff
@@ -137,7 +142,25 @@ private void writeTarget()
             try
             {
                 final HttpEntity entity = response.getEntity();
-
+                /*
+                * If the server responded with 200 but the content length doesn't match,
+                * we will delete the file and return a 502 Gateway Error status to indicate
```

Comment:
I think it would help for future reference to change this comment to something like "...we will delete the file and return a null Transfer as if the file wasn't found, since the content is invalid." You know, just because we moved away from the idea of returning a 502 for this...